core/dive.c

```diff
@@ -3728,13 +3730,11 @@ double depth_to_atm(int depth, const struct dive *dive)
  * take care of this, but the Uemis we support natively */
 int rel_mbar_to_depth(int mbar, const struct dive *dive)
 {
-	int cm;
-	double specific_weight = SEAWATER_SALINITY * 0.981 / 10000.0;
+	double specific_weight = SALINITY_TO_SPECIFIC_WEIGHT(SEAWATER_SALINITY);
 	if (dive->dc.salinity)
-		specific_weight = dive->dc.salinity / 10000.0 * 0.981;
+		specific_weight = SALINITY_TO_SPECIFIC_WEIGHT(dive->dc.salinity);
 	/* whole mbar gives us cm precision */
-	cm = (int)lrint(mbar / specific_weight);
-	return cm * 10;
+	return (int)lrint(mbar / specific_weight);
```

Comment:
I find this calculation really confusing.
```
1bar ~= 10m
1mbar ~= 10mm == 1cm
```
so that made sense. That extra factor of ten that you have in there... I must be missing something?